config/routes.rb

```diff
@@ -7,6 +7,7 @@
 
   post '/companies/update_companies_list', to: 'companies#update_companies_list'
   post '/companies/update_companies_details', to: 'companies#update_companies_details'
+  post '/companies/update_company_shareholders', to: 'companies#update_company_shareholders'
```

Comment:
why not pass the company ID this way?

```suggestion
  post '/companies/:id/update_company_shareholders', to: 'companies#update_company_shareholders'
```